root/programs/C#/Frameworks/Infrastructure/Framework/Util/GetMessage.cs

```diff
@@ -119,307 +107,179 @@ public static string GetMessageDescription(string messageID)
-            //bool xmlfilenotExist = false;
-            notExist = false;
-
-            string tempXMLFileName = string.Empty;
-            string tempXMLFilePath = string.Empty;
+            // defaultFile
+            string defaultFilePath = GetConfigParameter.GetConfigValue(FxLiteral.XML_MSG_DEFINITION);
+            string defaultFileName = "";
 
-            string defaultXMLFileName = string.Empty;
-            string defaultXMLFilePath = string.Empty;
-            string cultureWiseXMLFileName = string.Empty;
-            string cultureWiseXMLFilePath = string.Empty;
+            // culture
+            CultureInfo currentUICulture = null;
+            string uICultureName = "";
 
-            string cultureWiseXMLParentFileName = string.Empty;
-            string cultureWiseXMLParentFilePath = string.Empty;
+            // cultureWiseFile
+            string cultureWiseFilePath = "";
+            string cultureWiseFileName = "";
 
-            bool defaultXMLFilePath_ResourceLoaderExists = false;
-            bool defaultXMLFilePath_EmbeddedResourceLoaderExists = false;
-            bool cultureWiseXMLFilePath_ResourceLoaderExists = false;
-            bool cultureWiseXMLFilePath_EmbeddedResourceLoaderExists = false;
+            string content = "";
 
-            #endregion
-
-            defaultXMLFilePath = GetConfigParameter.GetConfigValue(FxLiteral.XML_MSG_DEFINITION);
-
-            GetMessage.GetCultureWiseXMLFileName(
-                cultureName, defaultXMLFilePath, out defaultXMLFileName, out cultureWiseXMLFilePath, out cultureWiseXMLFileName);
-
-            // This has been added for Threadsafe
-            lock (thisLock)
+            lock (GetMessage._lock)
             {
-                #region ContainsKey
-                
-                // Check that XML file is already loaded.
-                if (GetMessage.DicMSG.ContainsKey(cultureWiseXMLFileName))
+                do
                 {
-                    // culture wise XML file is already loaded.
-                    if (GetMessage.DicMSG[cultureWiseXMLFileName].ContainsKey(messageID))
+                    if (currentUICulture == null)
                     {
-                        return GetMessage.DicMSG[cultureWiseXMLFileName][messageID];
+                        // 初回
+                        currentUICulture = uiCulture;
                     }
                     else
                     {
-                        return string.Empty;
+                        // フォールバック
+                        currentUICulture = currentUICulture.Parent;
                     }
-                }
-                else
-                {
-                    // culture wise XML file isn't loaded.
-                }
 
-                if (GetMessage.DicMSG.ContainsKey(defaultXMLFileName))
-                {
-                    // default XML file is already loaded.
+                    // uICultureName
+                    uICultureName = currentUICulture.Name;
```

Comment:
@daisukenishino 

When `currentUICulture` is the *invariant culture*, `currentUICulture.Name` returns *string.Empty*. So, I think that it is unnecessary to execute `GetMessage.GetCultureWiseFile` method.